include/mxnet/c_api.h

```diff
@@ -55,7 +55,9 @@ extern "C" {
 #endif
 
 /*! \brief manually define unsigned int */
-typedef unsigned int mx_uint;
+typedef uint32_t mx_uint;
+/*! \brief manually define 64-bit int */
+typedef int64_t mx_int64;
```

Comment:
@larroy in language bindings we have created typedefs so that the signature of APIs on both sides(for e.g. C++ and python) are identical. Let me know if that answers your question

Reply:
@larroy I also feel this typedef with the rest mx_* typedef should go away in this file. @access2rohit has agreed to do it in his next PR for consistency purpose.